src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -252,11 +252,16 @@ private static bool CheckParameters(ISet<UDFArg> args, List<TexlError> errors, I
                 {
                     argsAlreadySeen.Add(arg.NameIdent.Name);
 
-                    if (!nameResolver.LookupType(arg.TypeIdent.Name, out var parameterType) || UserDefinitions.RestrictedTypes.Contains(parameterType._type))
+                    if (!nameResolver.LookupType(arg.TypeIdent.Name, out var parameterType))
                     {
                         errors.Add(new TexlError(arg.TypeIdent, DocumentErrorSeverity.Severe, TexlStrings.ErrUDF_UnknownType, arg.TypeIdent.Name));
                         isParamCheckSuccessful = false;
                     }
+                    else if (IsRestrictedType(parameterType))
+                    {
+                        errors.Add(new TexlError(arg.TypeIdent, DocumentErrorSeverity.Severe, TexlStrings.ErrUDF_InvalidParamType, arg.TypeIdent.Name));
```

Comment:
>ErrUDF_InvalidParamType [](http://example.com/codeflow?start=106&length=23)

The error message ("The stated function return type is invalid.") doesn't take any arguments, but we are passing one here. Passing the argument name sounds like the right thing to do, so that the maker will know which argument has a problem, so we should consider adding an argument (e.g., '{0}') to the resource string.